### PR TITLE
Add support for IAddressOf to control flow graph

### PR DESCRIPTION
**Pull request by chborl**

### CODE REVIEW

**Comment by chborl** [comment]
@AlekseyTs @333fred @mavasani @jaredpar for review.

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAddressOfOperation.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32*) (Syntax: 'p3 = &(x ? p1 : p2)->i')
+              Left: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.Int32*, IsImplicit) (Syntax: 'p3')
+              Right: 
+                IAddressOfOperation (OperationKind.AddressOf, Type: System.Int32*) (Syntax: '&(x ? p1 : p2)->i')
+                  Reference: 
+                    IFieldReferenceOperation: System.Int32 S2.i (OperationKind.FieldReference, Type: System.Int32) (Syntax: '(x ? p1 : p2)->i')
+                      Instance Receiver: 
+                        IOperation:  (OperationKind.None, Type: null, IsImplicit) (Syntax: '(x ? p1 : p2)')
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAddressOfOperation.cs]: >Type: null, [](start = 58, length = 11)

Why is the type `null` here? I would expect `S2*`

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAddressOfOperation.cs]: > Why is the type null here? I would expect S2*

Anything is a fair game for OperationKind.None. It means that whatever bound node we ran into is not handled by IOperation factory yet. There are a lot of not implemented things in the factories.

---
In reply to: [188758023](https://github.com/dotnet/roslyn/pull/26869#discussion_r188758023) [](ancestors = 188758023)

Comment (jaredpar) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAddressOfOperation.cs]: Is the expectation that when this work is done the type here will be `S2*`? Maybe my mental model is still off here but my intuition expects to see somewhat of a consistent type flow here. The insertion of `null` strikes me as off. 

---
In reply to: [188761551](https://github.com/dotnet/roslyn/pull/26869#discussion_r188761551) [](ancestors = 188761551,188758023)

**Pull request merged**